src/rust/engine/graph/src/lib.rs

```diff
@@ -338,7 +338,7 @@ impl<N: Node> InnerGraph<N> {
       .walk(
         root_ids.iter().cloned().collect(),
         Direction::Incoming,
-        |n| !n.cacheable(),
+        |id, graph| !graph.entry_for_id(*id).unwrap().node().cacheable(),
```

Comment:
Hm... was `self.entry_for_id` not working here? The `walk` doesn't take ownership of the graph, so it should be possible to just use it from the closure.

Not a blocker, just wondering.

Reply:
Oh, um, I assumed there would be some nasty lifetimes issue here with borrowing the reference to self in the closure, which was already taken care of in the walk function.

Reply:
Apparently your suggestion just works, so I'll change it.